ipq40xx: add support for PanguBox PGB-M5

lintel opened the pull request:
SOC:      IPQ4019 / QCA Dakota
CPU:      Quad-Core ARMv7 Processor rev 5 (v7l) Cortex-A7
DRAM:     512 MiB
NOR:      16 MiB  (Bootloader and ART)
NAND:     128 MiB (UBI for Firmware)
ETH:      Qualcomm Atheros QCA8075 (5 ports)
USB:      1 x 3.0 (Host controller in the SoC)
WLAN1:    Qualcomm Atheros QCA4019 2.4GHz 802.11bgn 2:2x2
WLAN2:    Qualcomm Atheros QCA4019 5GHz 802.11a/n/ac 2:2x2
M.2 slot: Qualcomm Atheros QCA9984 5GHz 802.11a/n/ac 4:4x4 WLAN Card
INPUT:    RESET Button
LEDS:     SYS, WiFi-2G, WiFi-5G, INTERNET, USB

Firmware upgrade step:
1.Press Reset button and power on the board,it will enter pb-boot recovery mode when SYS-LED blink.
2.Use web browser vist: http://192.168.1.1/index.html
3.Select firmware file and upload it.(support nand-factory.bin and sysupgrade.bin)

adschm commented:
Thanks for your contribution. It would be nice if you didn't spam with three PRs for one submission next time.

adschm commented on package/firmware/ipq-wifi/Makefile:
```diff
@@ -36,6 +36,7 @@ ALLWIFIBOARDS:= \
 	engenius_eap2200 \
 	engenius_emd1 \
 	engenius_emr3500 \
+	pangubox_pbg-m5 \
```
Please generally care about alphabetic sorting.

adschm commented on target/linux/ipq40xx/base-files/etc/board.d/01_leds:
```diff
@@ -46,6 +46,11 @@ engenius,ens620ext)
+pangubox,pgb-m5)
+	ucidef_set_led_wlan "wlan2g" "WLAN2G" "pgb-m5:2g" "phy0tpt"
+	ucidef_set_led_wlan "wlan5g" "WLAN5G" "pgb-m5:5g" "phy1tpt"
+	ucidef_set_led_usbport "usb" "USB" "pgb-m5:usb" "usb1-port1" "usb2-port1"
```
Can you use DT triggers instead?

lintel replied:
I have updated the dts，thx.

adschm commented on target/linux/ipq40xx/base-files/etc/board.d/02_network:
```diff
@@ -132,6 +133,11 @@ ipq40xx_setup_macs()
 		;;
+	pangubox,pgb-m5)
+		wan_mac=$(mtd_get_mac_binary ART 0)
+		lan_mac=$(mtd_get_mac_binary ART 6)
```
This can be done in DTS.

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-pangubox-pgb-m5.dts:
```diff
@@ -0,0 +1,343 @@
+		pinctrl-names = "default";
+
+		led_sys:sys {
+			label = "pgb-m5:sys";
```
Please use proper "color:function" scheme.

lintel replied:
It's done in new commit,thx

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-pangubox-pgb-m5.dts:
```diff
@@ -0,0 +1,343 @@
+
+			partition@3 {
+				label = "unused";
+				reg = <0x180000 0>;
```
The partitioning looks broken.

981213 replied:
A zero partition size tells kernel to expand this partition to the end of the flash, so it's not technically broken..
Whether we should allow using this is another question though.

adschm replied:
Broken was mostly referring to the overlapping partitions before.

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-pangubox-pgb-m5.dts:
```diff
@@ -0,0 +1,343 @@
+
+			ubi_part: partition@1 {
+				label = "ubi";
+				reg = <0x0 0>;
```
please define the partition size.

adschm commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -218,6 +218,24 @@ define Device/avm_fritzbox-4040
 endef
 TARGET_DEVICES += avm_fritzbox-4040
 
+define Device/pangubox-pgb-m5
```
Please use underscore after vendor name (will require adjustment of DTS name).

adschm commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -218,6 +218,24 @@ define Device/avm_fritzbox-4040
+	FILESYSTEMS := squashfs
+	IMAGES := nand-factory.bin nand-sysupgrade.bin
+	IMAGE/nand-factory.bin := append-ubi
+	DEVICE_PACKAGES :=  -kmod-ath10k-ct kmod-ath10k-ct-smallbuffers ath10k-firmware-qca9984-ct  \
```
Why smallbuffers?

lintel replied:
It was modified to ath10k-ct in new commit.

lintel committed:
ipq40xx: add support for PanguBox PGB-M5

SOC:      IPQ4019 / QCA Dakota
CPU:      Quad-Core ARMv7 Processor rev 5 (v7l) Cortex-A7
DRAM:     512 MiB
NOR:      16 MiB  (Bootloader and ART)
NAND:     128 MiB (UBI for Firmware)
ETH:      Qualcomm Atheros QCA8075 (5 ports)
USB:      1 x 3.0 (Host controller in the SoC)
WLAN1:    Qualcomm Atheros QCA4019 2.4GHz 802.11bgn 2:2x2
WLAN2:    Qualcomm Atheros QCA4019 5GHz 802.11a/n/ac 2:2x2
M.2 slot: Qualcomm Atheros QCA9984 5GHz 802.11a/n/ac 4:4x4 WLAN Card
INPUT:    RESET Button
LEDS:     SYS, WiFi-2G, WiFi-5G, INTERNET, USB

Firmware upgrade step:
1.Press Reset button and power on the board,it will enter pb-boot recovery mode when SYS-LED blink.
2.Use web browser vist: http://192.168.1.1/index.html
3.Select firmware file and upload it.(support nand-factory.bin and sysupgrade.bin)

Signed-off-by: lintel <lintel.huang@gmail.com>

lintel commented:
> Thanks for your contribution. It would be nice if you didn't spam with three PRs for one submission next time.

Sorry,  I will be careful on next time. :-)

lintel commented:
ping @981213 @adschm ,pls review the new commit,thx.

981213 commented:
FYI: We only use GitHub to post review comments and all the merge-related features on GitHub are unused here. So the "Review required", "pending reviewers" and "Merging is blocked" you see on the page mean nothing.

lintel commented:
ping @chunkeey Pls help review the code,thx.

lintel committed:
Merge branch 'master' into support-pgb-m5-board

lintel commented:
I will not contribute any code to this project. thanks a lot @981213
